deluge/core/alertmanager.py

```diff
@@ -60,12 +62,39 @@ def update(self):
         self.delayed_calls = [dc for dc in self.delayed_calls if dc.active()]
         self.handle_alerts()
 
+    def start(self):
+        thread = threading.Thread(target=self.wait_for_alert_in_thread, name="alert-poller")
+        thread.daemon = True
+        thread.start()
+
     def stop(self):
         for delayed_call in self.delayed_calls:
             if delayed_call.active():
                 delayed_call.cancel()
         self.delayed_calls = []
 
+    def wait_for_alert_in_thread(self):
+        while self._component_state not in ("Stopping", "Stopped"):
+            try:
```

Comment:
```suggestion
            if self.session.wait_for_alert(1000) is None:
                continue
```

Reply:
We should crash if there is an exception raised here. Alerts from libtorrent are critical for Deluge to function.

Reply:
Done.